src/main/java/cceclipseplugin/core/EclipseRequestManager.java

```diff
@@ -196,19 +196,16 @@ public void finishMoveFile(FileMetadata fMeta) {
 	}
 	
 	public void pullDiffSendChanges(FileMetadata fMeta) {
-		try {
-			Thread.sleep(1000);
-		} catch (InterruptedException e1) {
-			e1.printStackTrace();
-		}
 		MetadataManager mm = PluginManager.getInstance().getMetadataManager();
-		ProjectMetadata pMeta = mm.getProjectMetadata(mm.getProjectIDForFileID(fMeta.getFileID()));
-		String projLocation = mm.getProjectLocation(mm.getProjectIDForFileID(fMeta.getFileID()));
-		IPath filePath = new Path(fMeta.getRelativePath());
-//		filePath = filePath.append(fMeta.getFilename());
-		System.out.println("Project location: " + projLocation);
+		long fileID = fMeta.getFileID();
+		long projectID = mm.getProjectIDForFileID(fileID);
+		ProjectMetadata pMeta = mm.getProjectMetadata(projectID);
+		String projLocation = mm.getProjectLocation(projectID);
+		IPath filePath = new Path(fMeta.getFilePath());
 		System.out.println("pulldiffsendchanges for " + filePath);
-		IFile file = ResourcesPlugin.getWorkspace().getRoot().getFileForLocation(filePath);
+		IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(pMeta.getName());
+		System.out.println("Project location: " + projLocation);
```

Comment:
Could you clean up some of these print statements? If you want to leave some in, that's fine, but maybe condense them into one more descriptive line so everyone doesn't get spammed when debugging.